.github/actions/setup-rust/rust.json

```
@@ -0,0 +1,31 @@
+{
+    "problemMatcher": [
+        {
+            "owner": "rust",
+            "pattern": [
+                {
+                    "regexp": "^(warning|warn|error)(\\[(.*)\\])?: (.*)$",
+                    "severity": 1,
+                    "message": 4,
+                    "code": 3
+                },
+                {
+                    "regexp": "^([\\s->=]*(.*):(\\d*):(\\d*)|.*)$",
+                    "file": 2,
+                    "line": 3,
+                    "column": 4
+                }
```

```suggestion
                {
                    "regexp": "^([\\s->=]*(.*):(\\d*):(\\d*)|.*)$",
                    "file": 2,
                    "line": 3,
                    "column": 4
                },
                {
                    "regexp": "^\\d* *(\\||=)(.*)$",
                    "message": 0,
                    "loop": true
                }
```

I wanted it to be this, but it's not possible due to https://github.com/actions/toolkit/issues/319 (which should probably be re-opened or re-issued)

---

technically, this would be possible if it was possible for cargo to output json diagnostics and human diagnostics, the error annotations in the log would be strange however.